src/dos/dos.cpp

```diff
@@ -47,6 +48,138 @@ void DOS_SetError(Bit16u code) {
 	dos.errorcode=code;
 }
 
+int countries_date_format_yyyymmdd[] = {
+        2,   // Canadian-French
+        36,  // Hungary
+        38,  // Croatia
+        40,  // Romania
+        42,  // Czech Republic / Slovakia
+        46,  // Sweden
+        48,  // Poland
+        81,  // Japan
+        82,  // South Korea
+        86,  // China
+        354, // Iceland
+        886, // Taiwan
+};
+
+int countries_date_separator_slash[] = {
+        3,   // Latin America
+        30,  // Greece
+        32,  // Belgium
+        34,  // Spain
+        39,  // Italy
+        44,  // United Kingdom
+        55,  // Brazil
+        90,  // Turkey
+        785, // Arabic countries
+        886, // Taiwan
+        972, // Israel
+};
+
+int countries_date_separator_period[] = {
+        7,   // Russia
+        33,  // France
+        41,  // Switzerland
+        43,  // Austria
+        47,  // Norway
+        49,  // Germany
+        86,  // China
+        358, // Finland
+};
+
+int countries_time_separator_period[] = {
+        39,  // Italy
+        45,  // Denmark
+        46,  // Sweden
+        358, // Finland
+};
+
+int countries_decimal_separator_comma[] = {
+        1,   // United States
+        44,  // United Kingdom
+        61,  // International English
+        81,  // Japan
+        82,  // South Korea
+        86,  // China
+        886, // Taiwan
+};
+
+int countries_decimal_separator_period[] = {
+        3,   // Latin America
+        31,  // Netherlands
+        34,  // Spain
+        38,  // Yugoslavia
+        39,  // Italy
+        40,  // Romania
+        45,  // Denmark
+        46,  // Sweden
+        49,  // Germany
+        55,  // Brazil
+        351, // Portugal
+        354, // Iceland
+        785, // Arabic countries
+        972, // Israel
+};
+
+void DOS_SetCountry(uint16_t countryNo)
+{
+	if (dos.tables.country == NULL)
+		return;
+
+	// For US, Latin America and International English use 12h clock
+	*(dos.tables.country +
+	  17) = countryNo == 1 || countryNo == 3 || countryNo == 61 ? 0 : 1;
+
+	// Date format
+	if (countryNo == 1)          // United States
+		*dos.tables.country = 0; // MM-DD-YYYY
+	else if (std::find(std::begin(countries_date_format_yyyymmdd),
+	                   std::end(countries_date_format_yyyymmdd),
+	                   countryNo) != std::end(countries_date_format_yyyymmdd))
+		*dos.tables.country = 2; // YYYY-MM-DD
+	else
+		*dos.tables.country = 1; // DD-MM-YYYY
+
+	// Date separation character
+	if (std::find(std::begin(countries_date_separator_slash),
+	              std::end(countries_date_separator_slash),
+	              countryNo) != std::end(countries_date_separator_slash))
+		*(dos.tables.country + 11) = 0x2f; // Forward-slash (/)
+	else if (std::find(std::begin(countries_date_separator_period),
+	                   std::end(countries_date_separator_period), countryNo) !=
+	         std::end(countries_date_separator_period))
+		*(dos.tables.country + 11) = 0x2e; // Period (.)
```

Comment:
Suggest a similar enum class to handle all the separator types and their hex values.

Reply:
Done.